src/bespokelabs/curator/prompter/prompter.py

```diff
@@ -86,32 +91,49 @@ def __init__(
             model_name, prompt_func, parse_func, response_format
         )
         self.batch_mode = batch
-        if batch:
-            if batch_size is None:
-                batch_size = 1_000
-                logger.info(
-                    f"batch=True but no batch_size provided, using default batch_size of {batch_size:,}"
+        self.backend = backend
+        # Select request processor based on backend
+        if backend == "openai":
+            if batch:
+                if batch_size is None:
+                    batch_size = 1_000
+                    logger.info(
+                        f"batch=True but no batch_size provided, using default batch_size of {batch_size:,}"
+                    )
+                self._request_processor = OpenAIBatchRequestProcessor(
+                    model=model_name,
+                    batch_size=batch_size,
+                    temperature=temperature,
+                    top_p=top_p,
+                    presence_penalty=presence_penalty,
+                    frequency_penalty=frequency_penalty,
                 )
-            self._request_processor = OpenAIBatchRequestProcessor(
-                model=model_name,
-                batch_size=batch_size,
-                temperature=temperature,
-                top_p=top_p,
-                presence_penalty=presence_penalty,
-                frequency_penalty=frequency_penalty,
-            )
-        else:
-            if batch_size is not None:
+            else:
+                if batch_size is not None:
+                    logger.warning(
+                        f"Prompter argument `batch_size` {batch_size} is ignored because `batch` is False"
+                    )
+                self._request_processor = OpenAIOnlineRequestProcessor(
+                    model=model_name,
+                    temperature=temperature,
+                    top_p=top_p,
+                    presence_penalty=presence_penalty,
+                    frequency_penalty=frequency_penalty,
+                )
+        elif backend == "litellm":
+            if batch:
                 logger.warning(
-                    f"Prompter argument `batch_size` {batch_size} is ignored because `batch` is False"
+                    "Batch mode is not supported with LiteLLM backend, ignoring batch=True"
                 )
-            self._request_processor = OpenAIOnlineRequestProcessor(
+            self._request_processor = LiteLLMOnlineRequestProcessor(
                 model=model_name,
                 temperature=temperature,
```

Comment:
How are we thinking about supporting the other generation config parameters? [#62](https://github.com/bespokelabsai/curator/issues/62)

Reply:
Do we want the user to be able to pass in a litellm object? Would this be possible with our implementation (I may come back and answer my own question as I read on :-)

Reply:
we can support other generation config params but i don't think it's super urgent for this PR

Reply:
ok we will come back to it later